twoliter/embedded/rpm2img

```diff
@@ -442,26 +442,10 @@ resize2fs -M "${ROOT_IMAGE}"
 dd if="${ROOT_IMAGE}" of="${OS_IMAGE}" conv=notrunc bs=1M seek="${partoff[ROOT-A]}"
 
 # BOTTLEROCKET-VERITY-A
-veritypart_mib="${partsize[HASH-A]}"
-truncate -s "${veritypart_mib}M" "${VERITY_IMAGE}"
-veritysetup_output="$(veritysetup format \
-    --format "$VERITY_VERSION" \
-    --hash "$VERITY_HASH_ALGORITHM" \
-    --data-block-size "$VERITY_DATA_BLOCK_SIZE" \
-    --hash-block-size "$VERITY_HASH_BLOCK_SIZE" \
-    "${ROOT_IMAGE}" "${VERITY_IMAGE}" \
-    | tee /dev/stderr)"
-verityimage_size="$(stat -c %s "${VERITY_IMAGE}")"
-veritypart_bytes="$((veritypart_mib * 1024 * 1024))"
-if [ "${verityimage_size}" -gt "${veritypart_bytes}" ] ; then
-    echo "verity content is larger than partition (${veritypart_mib}M)"
-    exit 1
-fi
-VERITY_DATA_4K_BLOCKS="$(grep '^Data blocks:' <<<"${veritysetup_output}" | awk '{ print $NF }')"
-VERITY_DATA_512B_BLOCKS="$((VERITY_DATA_4K_BLOCKS * 8))"
-VERITY_ROOT_HASH="$(grep '^Root hash:' <<<"${veritysetup_output}" | awk '{ print $NF }')"
-VERITY_SALT="$(grep '^Salt:' <<<"${veritysetup_output}" | awk '{ print $NF }')"
-veritysetup verify "${ROOT_IMAGE}" "${VERITY_IMAGE}" "${VERITY_ROOT_HASH}"
+generate_verity "${ROOT_IMAGE}" "${VERITY_IMAGE}" "${partsize[HASH-A]}" \
+  VERITY_VERSION VERITY_HASH_ALGORITHM VERITY_DATA_BLOCK_SIZE \
+  VERITY_HASH_BLOCK_SIZE VERITY_DATA_4K_BLOCKS VERITY_DATA_512B_BLOCKS \
+  VERITY_ROOT_HASH VERITY_SALT
 dd if="${VERITY_IMAGE}" of="${OS_IMAGE}" conv=notrunc bs=1M seek="${partoff[HASH-A]}"
 
 declare -a DM_VERITY_ROOT
```

Comment:
The reason `generate_verity` isn't `generate_verity_root` is because `img2img` needs the individual values to avoid issues related to order adjustments.